src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> originalNode = getNodeByIndex(index);
+        T oldValue = originalNode.item;
+        originalNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T removedElement;
+        if (index == 0) {
+            removedElement = first.item;
+            first = first.next;
+            if (first == null) {
+                last = null;
+            }
+        } else {
+            Node<T> prev = getNodeByIndex(index - 1);
+            removedElement = prev.next.item;
+            prev.next = prev.next.next;
+            if (index == size - 1) {
+                last = prev;
+            }
+        }
+        size--;
+        return removedElement;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = getIndex(object);
+        if (index == -1) {
+            return false;
+        } else {
+            remove(index);
+            return true;
+        }
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void linkLast(T e) {
+        final Node<T> lastNode = last;
+        final Node<T> newNode = new Node<>(lastNode, e, null);
+        last = newNode;
+        if (lastNode == null) {
+            first = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        size++;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("The index " + index + " is invalid "
+                    + "the actual size of array is " + size);
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        checkIndex(index);
+        Node<T> current = first;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current;
+    }
+
+    private int getIndex(T object) {
+        Node<T> current = first;
+        int index = -1;
+        for (int i = 0; i < size; i++) {
+            if (current.item == object || current.item != null && current.item.equals(object)) {
+                index = i;
+                break;
+            }
+            current = current.next;
+        }
+        return index;
+    }
+
+    private static class Node<T> {
+        T item;
+        Node<T> next;
+        Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+
+        public Node(T value) {
+            this.item = value;
+        }
```

Comment:
Don't use getters and setters in class Node. Since `Node` is an inner class, you have access to private fields of the outer class `MyLinkedList` and don't need getters and setters.

Reply:
fix comment